app/models/filter.rb

```diff
@@ -61,7 +61,12 @@ def self.search_by_limited(key, operator, value)
     { :conditions => conditions }
   end
 
+  # This method attempts to return an existing class that is derived from the resource_type.
+  # In some instances, this may not be a real class (e.g. a typo) or may be nil in the case
+  # of a filter not having been saved yet and thus the permissions objects not being currently
+  # accessible.
   def self.get_resource_class(resource_type)
+    return nil if resource_type.nil?
```

Comment:
I guess it could hit this if you were defining permissions that weren't attached to a resource class or were more general/global permissions? If that situation can''t/shouldn't be happening then I'd need to look at preventing hitting this function period in the situations that lead to this (which I suppose would involve changing the ordering of events so that the permissions exist on the filter when this is called).